src/librustc/mir/interpret/allocation.rs

```diff
@@ -534,12 +560,90 @@ impl<'tcx, Tag, Extra> Allocation<Tag, Extra> {
             new_state,
         );
     }
+
+    pub fn compress_defined_range(
```

Comment:
Uh, given the PR title I had hoped this was just refactoring.  Please make sure your title reflects what you did!  Seems like beyond making `byte` private, this also changes the internal representaion?

This is two changes in one. Is there any chance you can split those into separate PRs? That would make them *a lot* easier to review.

Reply:
This just moves the compression from `interpret/memory.rs` to `allocation`. It doesn't (at least shouldn't) change its implementation. It's hard to hide `undef_mask` without moving this since `memory` accesses it extensively. Well, there could be an intermediate PR with `fn undef_mask_mut(&mut self)` but that didn't really seem sensible?

Reply:
Oh, I missed that this is moved in from there, sorry.

But then why the changes in the `Alloc` struct, the changed doc comment for `bytes` and the new `size` field?

Reply:
Probably missed them when I cherry-picked some of the accessor utilities from the other PR.. Just now realized that they include references to not allocating all of the memory. Sorry for the confusion.